mobile/lib/auth/domain/repositories/auth_repository.dart

```diff
@@ -0,0 +1,91 @@
+import 'package:google_sign_in/google_sign_in.dart';
+import 'package:mobile/auth/auth.dart';
+import 'package:sign_in_with_apple/sign_in_with_apple.dart';
+import 'package:supabase_flutter/supabase_flutter.dart';
+
+class AuthRepository {
+  AuthRepository({
+    required GoogleSignIn googleSignIn,
+    required GoogleSignInAccount googleSignInAccount,
+  }) {
+    _googleSignIn = googleSignIn;
+    _googleSignInAccount = googleSignInAccount;
+  }
+
+  final _supabase = Supabase.instance.client;
```

Comment:
It would be good to also inject this through the constructor. It helps with dependency injection and writing unit tests. (Yes, it's possible to test using zones but I think the dependency injection route is the more common approach so more devs will be familiar with it)

Reply:
Also since we'll only be using the auth and not other parts of the client, we can just have the variable be of type auth and not the supabase client